### PR TITLE
Add a metric for observing recommendation stability & changes.

### PR DESCRIPTION
**Pull request by kgolab**
The relative_recommendation_changes metric reports changes seen on any given recommendation.
This allows to observe stability vs noisiness of recommendations. Ideally the metric should report mostly zeros as we expect the recommendations to diverge with time and change only rarely.

### CODE REVIEW

**Comment by kgolab** [comment]
@bskiba , please take a look. This is the last PR in "metrics chain".

File: vertical-pod-autoscaler/pkg/utils/metrics/quality/quality.go [hunk 1]
```diff
@@ -184,6 +196,44 @@ func ObserveQualityMetricsRecommendationMissing(usage float64, isOOM bool, resou
...
+		klog.Warningf("Cannot compare with current recommendation being nil. VPA mode: %v, size: %v", updateMode, vpaSize)
+		return
+	}
+
+	for resource, amount := range current {
+		newValue := quantityAsFloat(resource, amount)
+		oldValue := quantityAsFloat(resource, previous[resource])
+
+		if oldValue > 0.0 {
+			diff := newValue/oldValue - 1.0 // -1.0 to fall into buckets from -1.0 to 100.0
```
Comment (bskiba) [inline vertical-pod-autoscaler/pkg/utils/metrics/quality/quality.go:216]: -1.0 to report decrease as negative value.

Comment (bskiba) [reply vertical-pod-autoscaler/pkg/utils/metrics/quality/quality.go:216]: (meaning can we clarify the intent more in the comment, fitting into buckets is just one of the reasons)

Comment (kgolab) [reply vertical-pod-autoscaler/pkg/utils/metrics/quality/quality.go]: Done.

File: vertical-pod-autoscaler/pkg/recommender/model/vpa.go [hunk 2]
```diff
@@ -142,15 +143,16 @@ func (vpa *Vpa) UseAggregationIfMatching(aggregationKey AggregateStateKey, aggre
 
 // UpdateRecommendation updates the recommended resources in the VPA and its
 // aggregations with the given recommendation.
-func (vpa *Vpa) UpdateRecommendation(recommendation *vpa_types.RecommendedPodResources) {
-	vpa.Recommendation = recommendation
+func (vpa *Vpa) UpdateRecommendation(recommendation *vpa_types.RecommendedPodResources, vpaSize int) {
```
[the remark this answers is not in the document]
Comment (kgolab) [reply vertical-pod-autoscaler/pkg/recommender/model/vpa.go]: You're right; I refactored VpaPodCount to be part of VPA object and everything seems much nicer now, thanks.

**Comment by bskiba** [comment]
Thanks, appreciate the descriptive comments 👍 
/lgtm
/approve

**Pull request merged**